### PR TITLE
Bug 1813877 - Announce page title with talkback when swiping on navbar

### PR DESCRIPTION
**Pull request by DreVla**
Talkback now announces the title of the tab switched to using swipe left or right gestures on the navbar.

https://user-images.githubusercontent.com/32488956/218774755-0680500f-657c-4b08-94e1-2ef3bd7ab62c.mp4

### Pull Request checklist

- [X] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [X] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1813877

### CODE REVIEW

**Review by Mugurell — APPROVED** [review]
Tested also and LGTM for the scenario of swiping between tabs.
Seeing that there are issues with what is read when a page is first opened, maybe this could warrant a new ticket.

File: android-components/components/browser/toolbar/src/main/res/layout/mozac_browser_toolbar_displaytoolbar.xml [hunk 1]
```diff
@@ -105,6 +105,7 @@
         android:layout_height="40dp"
         android:layout_marginTop="8dp"
         android:paddingEnd="@dimen/mozac_browser_toolbar_origin_padding_end"
+        android:accessibilityLiveRegion="polite"
```
Comment (DreVla) [inline android-components/components/browser/toolbar/src/main/res/layout/mozac_browser_toolbar_displaytoolbar.xml:108]: The previous implementation with `announceForAccessibility` was not the recommended one and thanks to @mcarare, who recommended me to use accessibility live region, I achieved a similar results but with a cleaner implementation. 
Video (there is no audio in the recording for some reason...):

https://user-images.githubusercontent.com/32488956/219055185-564fe7b8-9c7e-47e7-bb85-16d74f5e1679.mp4

Comment (Mugurell) [reply android-components/components/browser/toolbar/src/main/res/layout/mozac_browser_toolbar_displaytoolbar.xml:108]: With this approach we are reading the full url + "Search or enter address" as an action but without the user imediately being able to - other Views may be selected.
With the previous approach we would be reading the page title, the same as it happens on Chrome.
A bigger issue though is that with this approach everytime the page is scrolled the announcement is repeated. On a Samsung A53 with Android 12 I saw the same announcement repeated every ~20 seconds if leaving the device alone.

(the displayed text is what Talkback would read)

https://user-images.githubusercontent.com/11428869/219067330-a17212f0-e153-410b-881e-04868217c984.mp4

Comment (DreVla) [reply android-components/components/browser/toolbar/src/main/res/layout/mozac_browser_toolbar_displaytoolbar.xml]: Indeed, it seems like this approach would not fit in this case. I have reverted back to the previous one.

**Comment by DreVla** [comment]
After playing around a bit with `accessibilityLiveRegion` and a bit of help from Mihai, I have updated this PR with a better fix that does not include `announceForAccessibility` since it is more of a convenience function.

Here is a video, the behavior is similar as before, but I made sure there were no bugs encountered when scrolling or when the webview was not focused by Talkback

https://user-images.githubusercontent.com/32488956/219404997-2ef9da70-adab-4f98-bbf0-b77ced5422dd.mp4

File: fenix/app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt [hunk 2]
```diff
@@ -1143,6 +1146,7 @@ abstract class BaseBrowserFragment :
                 val toolbarHeight = resources.getDimensionPixelSize(R.dimen.browser_toolbar_height)
                 val context = requireContext()
                 resumeDownloadDialogState(selectedTab.id, context.components.core.store, context, toolbarHeight)
+                // it.announceForAccessibility(selectedTab.content.title)
```
Comment (Mugurell) [inline fenix/app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt:1149]: LGTM.
Don't forget to remove this.

**Comment by BlindMoon38** [comment]
Considering this is being uplifted, take a look at [Bug1817611](https://bugzilla.mozilla.org/show_bug.cgi?id=1817611), please. The timing of the worsening of the issue matches the merging of this PR.

CC: @DreVla @Mugurell.

**Comment by Mugurell** [comment]
> Considering this is being uplifted, take a look at [Bug1817611](https://bugzilla.mozilla.org/show_bug.cgi?id=1817611), please. The timing of the worsening of the issue matches the merging of this PR.
> 
> CC: @DreVla @Mugurell.

Thank you for the ping! Agree on the possible link between this patch using `accessibilityLiveRegion` and the issues from [Bug 1817611](https://bugzilla.mozilla.org/show_bug.cgi?id=1817611). I think we should prioritize investigating the Bugzilla ticket and fixup or backout this patch if needed.

**Pull request merged**